src/Google.Pubsub.V1/SimpleSubscription.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using static Google.Pubsub.V1.PublisherClient;
+using static Google.Pubsub.V1.SubscriberClient;
+
+
+namespace Google.Pubsub.V1
+{
+    public class SimpleSubscription
+    {
+        const int MaxMessages = 10;
+
+        public string ProjectId => Parent.ProjectId;
+        public string SubscriptionId { get; }
+        public string TopicId { get; }
+        public string TopicResourceName { get; }
+        public string ResourceName { get; }
+        public SimplePubsub Parent { get; }
+
+        public SimpleSubscription(SimplePubsub parent, string subscriptionId, string topicId)
+        {
+            Parent = parent;
+            SubscriptionId = subscriptionId;
+            TopicId = topicId;
+            ResourceName = GetSubscriptionName(parent.ProjectId, subscriptionId);
+            TopicResourceName = GetTopicName(parent.ProjectId, topicId);
+        }
+
+        // Options for completion:
+        // - CancellationToken
+        // - Return an IDisposable, and disposing of it stops (as per Rx)
+        // Other option: create another class which exposes an event. Or use IObservable.
+        // There are plenty of options for the async case, too...
+        // If we continue with the cancellation token, should we throw if the cancellation token is cancelled, or just return?
```

Comment:
If we do use cancellation token, then:
* For the async method, calling cancellationToken.ThrowIfCancellationRequested() is suitable, as this will leave the Task in a Cancelled state.
* For the sync method, I think we should simply return on cancellation, which is (in my opinion) more analogous to a Cancelled task than throwing an exception would be.
  Throwing an exception would also force user code to catch an exception in what probably isn't really an error situation.